R/tmiic.wrapper.R

```diff
@@ -0,0 +1,780 @@
+  # NB: for non lag 0 edges, such cases are more than likely errors
+  # and will generate negative lags however showing them will allow the user
+  # to identify possible issues
+  #
+  for (idx in 1:nrow(df) )
+    {
+    if (df[idx,"infOrt"] == -2)
+      df[idx, c("x","y","lag")] <- c (df[idx,"y"], df[idx,"x"],
+                                      -as.integer (df[idx,"lag"]) )
+
+    if ( (df[idx,"infOrt"] == 6) & (as.integer (df[idx,"lag"]) != 0) )
+      df[nrow(df)+1, c("x","y","lag")] <- c (df[idx,"y"], df[idx,"x"],
+                                               -as.integer (df[idx,"lag"]) )
+    }
+  #
+  # Combine lags from node1 < node2, lag0 and node1 >= node2
+  #
+  list_lag1 <- unique (df[ ( (df$x < df$y) & (df$lag != 0) ),]$lag)
+  list_lag2 <- unique (df[ (df$lag == 0),]$lag)
+  list_lag3 <- unique (df[ ( (df$x >= df$y) & (df$lag != 0) ),]$lag)
+
+  list_lag1 <- paste (unlist (list_lag1[order (list_lag1, decreasing=TRUE)]), collapse=",")
+  list_lag2 <- as.character (list_lag2)
+  list_lag3 <- paste (unlist (list_lag3[order (list_lag3)]), collapse=",")
+
+  list_lags <- c(list_lag1[[1]], list_lag2, list_lag3[[1]])
+  list_lags <- lapply (list_lags, function(z) { z[!is.na(z) & z != ""]})
+  if ( (list_lag1[[1]] != "") & (list_lag3[[1]] != "") )
+    list_lags <- paste (unlist (list_lags), collapse="/")
+  else
+    list_lags <- paste (unlist (list_lags), collapse=",")
+
+  return (list_lags)
+  }
+
+#-------------------------------------------------------------------------------
+# tmiic_combine_orient
+#-------------------------------------------------------------------------------
+# Utility function to combine edges orientations when flattening the network.
+#
+# params:
+# - df: a dataframe with the edges to combine
+# - col_name: string, the orientation column
+#-------------------------------------------------------------------------------
+tmiic_combine_orient   <- function (df, col_name)
+  {
+  df <- df[!is.na (df[[col_name]]),]
+  if (nrow (df) <= 0)
+    return (NA)
+  #
+  # We set orientations as if node X <= node Y
+  # NB: we do not care of x, y, lag and proba columns as they are not used
+  # later in the function
+  #
+  for (idx in 1:nrow(df) )
+    if ( (df[idx,"x"] > df[idx,"y"]) & (!is.na (df[idx, col_name])) )
+        if (abs(df[idx, col_name]) == 2)
+          df[idx, col_name] <- -(df[idx, col_name])
+
+  col_min <- min (df[, col_name])
+  col_max <- max (df[, col_name])
+
+  if ( (col_max == 6) | ((col_min == -2) & (col_max == 2)) )
+    return (6)
+  else if (col_max == 2)
+    return (2)
+  else if (col_min == -2)
+    return (-2)
+  else
+    return (1)
+  }
+
+#-----------------------------------------------------------------------------
+# tmiic_combine_probas
+#-----------------------------------------------------------------------------
+# Utility function to combine edges probabilities when flattening the network.
+# Depending on the combined edges orientation, chose the appropriate max, min
```

Comment:
EN (OD)

Reply:
Utility function to combine edge probabilities when flattening the network.
Depending on the combined edges orientation, chooses the appropriate max, min
# or mean probabilities to compute the combined edge probabilities